apps/api/src/applicant/ienapplicant.util.service.ts

```diff
@@ -63,43 +95,80 @@ export class IENApplicantUtilService {
       return this.ienapplicantRepository.findAndCount(query);
     }
 
-    let where: object = {};
-    let isWhere = false;
+    let isStatus = false;
+    let whereStatus: object;
     if (status) {
       const status_list = await this.fetchChildStatusList(status);
-      where = { status: In(status_list), ...where };
-      isWhere = true;
+      isStatus = true;
+      whereStatus = { status: In(status_list) };
     }
 
-    if (name && name.trim() !== '') {
-      where = { name: ILike(`%${name.trim()}%`), ...where };
-      isWhere = true;
+    let isName = false;
+    let whereName = '';
+    if (name) {
+      whereName = this._nameSearchQuery(name);
+      isName = true;
     }
 
     let isHaPcn = false;
+    let whereHa = '';
     if (ha_pcn) {
-      // check details to identify filter
-      isHaPcn = true;
-    }
-    if (isHaPcn) {
       const ha_pcn_array = ha_pcn?.split(',');
-      let ha_search_sql = ha_pcn_array
+      whereHa = ha_pcn_array
         ?.map(id => {
           return `health_authorities @> '[{"id":${id}}]'`;
         })
-        .join(' or ');
-      ha_search_sql = `(${ha_search_sql})`;
+        .join(' OR ');
+      whereHa = `(${whereHa})`;
+      isHaPcn = true;
+    }
 
+    if (isName && isStatus && isHaPcn) {
```

Comment:
I tried this.
```typescript

    const conditions: (string|ObjectLiteral)[] = [];

    if (status) {
      const status_list = await this.fetchChildStatusList(status);
      conditions.push({ status: In(status_list) });
    }

    if (name) {
      conditions.push(this._nameSearchQuery(name));
    }

    if (ha_pcn) {
      const ha_pcn_array = ha_pcn?.split(',');
      const condition = ha_pcn_array
        ?.map(id => {
          return `health_authorities @> '[{"id":${id}}]'`;
        })
        .join(' OR ');
      conditions.push(`(${condition})`);
    }

    return this.ienapplicantRepository.findAndCount({
      where: (qb: SelectQueryBuilder<IENApplicant>) => {
        const condition = conditions.shift();
        if (condition) qb.where(condition);
        conditions.forEach(condition => qb.andWhere(condition));
      },
      ...query,
    });
```

Reply:
I was above pushing one simple way, but your approach looks cleaner. Much appreciated the suggestion here. As I was about to ask you for the same in our group.